yt/python/yt/wrapper/default_config.py

```diff
@@ -873,51 +880,128 @@ def _update_from_env_patch(config):
-    for opt_name in ["YT_CONFIG_PATH", "YT_CONFIG_FORMAT"]:
-        if opt_name in os.environ:
-            config[SHORTCUTS[opt_name[3:]]] = os.environ[opt_name]
-    config_path = config["config_path"]
-
-    if config_path is None:
-        home = common.get_home_dir()
-
-        config_path = "/etc/ytclient.conf"
-        if home:
-            home_config_path = os.path.join(os.path.expanduser("~"), ".yt/config")
-            if os.path.isfile(home_config_path):
-                config_path = home_config_path
-
+class ConfigParserV2:
+    VERSION = 2
+    _PROFILES_KEY = "profiles"
+    _DEFAULT_PROFILE_KEY = "default_profile"
+
+    def __init__(self, config, profile):
+        self._config = config
+        self._profile = profile
+
+    def extract(self):
+        if self._PROFILES_KEY not in self._config:
+            raise common.YtError("Missing {0} key in YT config".format(self._PROFILES_KEY))
+        profiles = self._config[self._PROFILES_KEY]
+        if not isinstance(profiles, dict):
+            raise common.YtError("Profiles should be dict, not {0}".format(type(profiles)))
+        current_profile = self._profile
+        if current_profile is None:
+            current_profile = self._config.get(self._DEFAULT_PROFILE_KEY)
+            if current_profile is None:
+                raise common.YtError("Profile has not been set and there is no default profile in the config")
+        if current_profile not in profiles:
+            raise common.YtError("Unknown profile {0}. Known profiles: {1}".format(
+                current_profile,
+                ",".join(profiles.keys())),
+            )
+        profile = profiles[current_profile]
+        return profile
+
+
+class _ConfigFSHelper:
+    """
+    Helper class for _update_from_file that will be overridden in tests.
+    """
+    def is_file(self, path):
+        return os.path.isfile(path)
+
+    def validate(self, path):
         try:
-            with open(config_path, "r"):
+            with open(path, "r"):
                 pass
+            return True
         except IOError:
-            config_path = None
-
-    if config_path and os.path.isfile(config_path):
-        load_func = None
-        format = config["config_format"]
-        if format == "yson":
-            load_func = yson.load
-        elif format == "json":
-            load_func = json.load
-        else:
-            raise common.YtError("Incorrect config_format '%s'" % format)
+            pass
+        return False
+
+    def get_home_dir(self):
+        return common.get_home_dir()
+
+    def read(self, path):
         try:
-            with open(config_path, "rb") as f:
-                common.update_inplace(config, load_func(f))
+            with open(path, "rb") as f:
+                return f.read()
         except Exception:
-            print("Failed to parse YT config from " + config_path, file=sys.stderr)
+            print("Failed to read YT config from " + path, file=sys.stderr)
```

Comment:
Мне кажется, странно печатать что-то из библиотеки в stderr.

Надо либо использовать логгер, либо оборачивать исключение.

Reply:
We can't use logging here because this code may be run in import time.

I see it's a common practice here to print error to stderr, but I can replace it by raising YtConfigError.

Reply:
yes, I see it now, 

I don't like it, but let's keep it to be consistent